eclair-core/src/main/scala/fr/acinq/eclair/message/OnionMessages.scala

```diff
@@ -94,6 +95,9 @@ object OnionMessages {
       1300
     } else if (payloadSize <= 32768) {
       32768
+    } else if (payloadSize > 65432) {
+      // A payload of size 65432 corresponds to a total lightning message size of 65535.
+      throw new Exception(s"Message is too large: payloadSize=$payloadSize")
```

Comment:
Note that this value will become incorrect once we start adding tlvs to the onion message's tlv stream. When we add tlvs to onion messages, we'll need to make sure we don't forget to update this code (we'll need to subtract the encoded length of the tlv stream).